jovo-framework/src/middleware/Handler.ts

```diff
@@ -385,11 +405,69 @@ export class Handler implements Plugin {
             }]);
 
             ComponentPlugin.initializeComponents(this.$handleRequest!);
-            const state = this.getState() ? `${this.getState()}.${componentName}` : componentName;
 
+            const state = this.getActiveComponentsRootState() ? `${this.getActiveComponentsRootState()}.${componentName}` : componentName;
+
+            this.triggerStateValidation = false;
             return this.toStateIntent(state, 'START');
         };
 
+        /**
+         * Checks if the given state contains the name of a initialized component.
+         * @throws {JovoError}
+         */
+        Jovo.prototype.checkStateForInitializedComponentName = function (state: string | undefined): void {
```

Comment:
How about adding  `checkStateForInitializedComponentName` to the Handler class as a private method?

Reply:
Moved it into the Handler class similar to the handle* methods, hope that's what you meant. Otherwise I'm not sure how to do "private" methods, I haven't had many visits to the TS-Camp yet.

Reply:
It was in the right place. Can you put it back into the Jovo class? :)
Sorry, my mistake.

Reply:
Moved it back to the Jovo class. Hope everything is alright then. Really looking forward to a release with these changes :)